### PR TITLE
feat: use full data uris

### LINKED ISSUE #194: change item image to be the full data uri instead of only the data part
currently it assumes all images are png or that the browser displays it anyway even when providing the wrong mime type:
`data:image/png;base64,${item.image!}`

It would be more clear/correct to put the mimetype and rest of the data url also into the `item.image` property.

That way you can use jpg, png and later even svg in uriencoding instead of base64.

### PR DESCRIPTION
**Pull request by Septias**
close #194

### CODE REVIEW

File: src/import.rs [hunk 1]
```diff
@@ -69,18 +69,18 @@ pub async fn import_many(
...
+                .map(|(index, entry)| {
+                    (index, entry.entry().filename().as_str().unwrap_or_default())
                 })
-                .map(|a| a.0);
-            let image = if let Some(index) = image {
+                .find(|(_, name)| *name == "icon.png" || *name == "icon.jpg");
+            let image = if let Some((index, name)) = image {
                 let res = read_vec(&reader, index).await?;
-                encode(&res)
+                let ending = name
```
Comment (r10s) [inline src/import.rs:78]:
```suggestion
                let extension = name
```

the `.png` or `.jpg` part of a file name is called "extension" or "suffix", i would stay with that well known wording (also change at the other places)

File: src/import.rs [hunk 2]
```diff
@@ -69,18 +69,18 @@ pub async fn import_many(
...
+                    .split('.')
+                    .nth(1)
+                    .context(format!("Can't extract file ending from {name}"))?;
+                let base64 = encode(&res);
+                format!("data:image/{ending};base64,{base64}")
```
Comment (r10s) [inline src/import.rs:83]: this is wrong. the mimetype for .jpg files is `image/jpeg`, see eg. https://www.w3.org/Graphics/JPEG/

although browser will probably correct the fault, i'd go for "be strict in what you write, be liberal in what you accept from others"

**Review by r10s — COMMENTED** [review]
moreover, i'd also use at least one image/jpeg in the mock data, so that all accepted image types are covered and one will see problems early

**Review by Simon-Laux — CHANGES_REQUESTED** [review]
https://github.com/webxdc/store/pull/199/files#diff-5d954d9a2e243edfbe0547bea7eb29755d24446dd2b527818d1f3fb6b4373b31R218
the download drag and drop still adds the prefix

File: src/import.rs [hunk 3]
```diff
@@ -69,18 +69,21 @@ pub async fn import_many(
...
-                .map(|a| a.0);
-            let image = if let Some(index) = image {
+                .find(|(_, name)| *name == "icon.png" || *name == "icon.jpg");
+            let image = if let Some((index, name)) = image {
                 let res = read_vec(&reader, index).await?;
-                encode(&res)
+                let mut extension = name
+                    .split('.')
+                    .nth(1)
+                    .context(format!("Can't extract file ending from {name}"))?;
```
Comment (r10s) [inline src/import.rs:81]:
```suggestion
                    .context(format!("Can't extract file extension from {name}"))?;
```

**Review by Simon-Laux — APPROVED** [review]
LGTM

**Pull request merged**